R/pkg/inst/tests/test_sparkSQL.R

```diff
@@ -294,9 +294,21 @@ test_that("create DataFrame with complex types", {
 test_that("create DataFrame from a data.frame with complex types", {
   ldf <- data.frame(row.names=1:2)
   ldf$a_list <- list(list(1, 2), list(3, 4))
+  ldf$an_envir <- c(as.environment(list(a=1, b=2)), as.environment(list(c=3)))
+
   sdf <- createDataFrame(sqlContext, ldf)
+  collected <- collect(sdf)
+
+  expect_equivalent(ldf[, 1, FALSE], collected[, 1, FALSE])
+
+  asSortedList <- function(e) {
+      tmp <- as.list(e)
+      tmp[sort(names(tmp))]
+  }
 
-  expect_equivalent(ldf, collect(sdf))
+  expect_equal(
```

Comment:
seems
expect_equivalent(ldf$an_envir, collected$an_envir) should be OK?
 asSortedList is not necessary?